### PR TITLE
[ALS-6248] Add package data stepper

### PR DESCRIPTION
**Pull request by JamesPeck**

### CODE REVIEW

File: src/lib/components/explorer/export/ExportStepper.svelte [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+  let statusPromise: Promise<string>;
+  let preparePromise: Promise<void>;
+  let datasetNameInput: string = '';
+  let picsureResultId: string = '';
+  let lockDownload = true;
+  let error: string = '';
+  let namedDataset: DataSet;
+  $: datasetId = '';
+  $: canDownload = true;
+  $: apiExport = false;
```
Comment (srpiatt) [inline src/lib/components/explorer/export/ExportStepper.svelte:33]: What sets this?

File: src/lib/components/explorer/export/ExportStepper.svelte [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+    try {
+      const res = await api.post(`picsure/query/${datasetId}/result`, {});
+      const responseDataUrl = URL.createObjectURL(new Blob([res], { type: 'octet/stream' }));
+      if (browser) {
+        const link = document.createElement('a');
+        link.href = responseDataUrl;
+        link.download = 'pic-sure-data.csv';
+        document.body.appendChild(link);
+        link.click();
+        document.body.removeChild(link);
```
Comment (srpiatt) [inline src/lib/components/explorer/export/ExportStepper.svelte:50]: This makes me sad but I guess I understand it.

File: src/lib/components/explorer/export/ExportStepper.svelte [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+      if (!apiExport) {
+        $state.current = 0;
+        goto('/dataset/' + namedDataset.uuid);
+      }
+    } catch (error) {
+      console.error('Error in onCompleteHandler', error);
+    }
+  }
+  function onNextHandler(e: CustomEvent): void {
+    console.log('event:next', e);
```
Comment (srpiatt) [inline src/lib/components/explorer/export/ExportStepper.svelte:61]: Don't forget to remove all these console logs for Part 2.

File: src/lib/components/explorer/export/ExportStepper.svelte [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+          if (status !== 'PENDING' || status !== 'RUNNING' || status !== 'QUEUED') {
+            clearInterval(interval);
+            if (status === 'ERROR') {
+              lockDownload = true;
+              reject(status);
+            } else {
+              lockDownload = false;
+              resolve(status);
+            }
+          }
```
Comment (srpiatt) [inline src/lib/components/explorer/export/ExportStepper.svelte:84]:
```suggestion
         if (status === 'ERROR') {
            lockDownload = true;
            clearInterval(interval);
            reject(status);
          } else if (!['PENDING', 'RUNNING', 'QUEUED'].includes(status)) {
            lockDownload = false;
            clearInterval(interval);
            resolve(status);
          }
```
This looks a bit clearer to me, even if it's slightly redundant in one place.

**Review by srpiatt — APPROVED** [review]

File: src/lib/components/explorer/export/ExportStepper.svelte (hunk 1 above)
Comment (JamesPeck) [reply src/lib/components/explorer/export/ExportStepper.svelte:33]: Its supposed to be the features from .env but We have not gotten that far yet. They don't need to be reactive either.

**Pull request merged**